### ISSUE TITLE
fix(types): allow empty string for lang attribute (BCP47)

### ISSUE THREAD

**Issue by YusukeHirao** [markuplint/markuplint#3585]
## Describe the bug

The `invalid-attr` rule reports `lang=""` and `<html lang>` (valueless = empty string) as a BCP47 violation. However, the HTML Living Standard explicitly allows an empty `lang` attribute to indicate "unknown language".

> If the attribute's value is the empty string, the language is set to unknown.
> — https://html.spec.whatwg.org/multipage/dom.html#the-lang-and-xml:lang-attributes

## Code Example

```html
<!doctype html><html lang><meta charset=utf-8>
<title>test</title>
```

## The current behavior

```
[1:26] (invalid-attr) The "lang" attribute must not be empty. It expects the BCP47 format
```

## The expected behavior

No violations reported.

## Impact

Causes ~45 false positives in the nu-html-checker compatibility benchmark.

## Affected package

`@markuplint/types` — BCP47 type checker
